### PR TITLE
contributor guide: document the revert process

### LINKED ISSUE #1809: Documenting the revert process (Avoid the github revert button)
- Use `git revert SHA` where SHA is the hash of the commit you wish to revert
- Push this commit to a new branch in your own repo
- Use the commit/branch to file a PR

Essentially avoid pushing the button in github web UI. This creates unnecessary branches in the kubernetes main repository that someone has to go clean up. The bots notice such merges and notifies kubernetes-dev channel as well. following process above will avoid that notifications too

### PR DESCRIPTION
**Pull request by nikhita**
Fixes #1809 

/cc cblecker dims 
/assign castrojo

Not sure if this is the best place to add it but I guess it comes under the github workflow?

### CODE REVIEW

**Comment by idvoretskyi** [comment]
/hold

for other reviewers

**Comment by parispittman** [comment]
Good tip, @nikhita. This may make better sense in a developers guide but TBD on that. Will wait for @castrojo and @guineveresaenger to weigh in.

**Comment by nikhita** [comment]
> This may make better sense in a developers guide

Yes, this sounds better! :+1:

**Comment by guineveresaenger** [comment]
I actually like it in the contributor guide. I don't imagine it coming up particularly often for a new contributor (and the caveat only applies to people senior enough with write access), but it still enshrines ideal and polite behavior AKA always operate on your own fork.

**Comment by cblecker** [comment]
@nikhita This is good. The one thing to possibly add is how to pick the commit hash (single commit, vs reverting an entire merge commit).

File: contributors/guide/github-workflow.md [hunk 1]
```diff
@@ -235,3 +235,25 @@ git rebase -i upstream/master
...
+### Reverting a commit
+
+In case you wish to revert a commit, use the following instructions.
+
+_If you have upstream write access_, please refrain from using the
+`Revert` button in the GitHub UI for creating the PR, because GitHub
+will create the PR branch inside the main repository rather than inside your fork.
+
+```sh
+# create a branch
```
Comment (mikedanese) [inline contributors/guide/github-workflow.md]: Add a suggestion to start from a synced master branch?

**Comment by idvoretskyi** [comment]
Any objections here to move forward with the PR? It's active for around one month.

**Comment by nikhita** [comment]
> Any objections here to move forward with the PR? 

Sorry, this slipped under my radar. I'll add the details that were suggested. :+1:

**Comment by dims** [comment]
/uncc @dims 

please add me back when you are ready @nikhita

**Comment by cblecker** [comment]
Hey @nikhita.. are you still able to finish this? Or would you like help to carry it through to completion?

**Comment by guineveresaenger** [comment]
@nikhita @cblecker making a bit of noise here as I was going through the outstanding work

**Comment by nikhita** [comment]
I am really sorry for letting this stay open for so long. I was busy with exams back then but I've graduate now :slightly_smiling_face: and have been back for a couple of weeks. Should have got back to this sooner.

> The one thing to possibly add is how to pick the commit hash (single commit, vs reverting an entire merge commit).

@cblecker for a merge commit, `git revert -m 1 <merge-commit>` would be the way to go?

**Comment by guineveresaenger** [comment]
@nikhita - no worries at all, that's why we have a community.
Congratulations on graduating!!!

**Comment by idvoretskyi** [comment]
@nikhita can you please update the PR with the latest comments addressed?

File: contributors/guide/github-workflow.md (hunk 1 above)
Comment (nikhita) [reply contributors/guide/github-workflow.md]: > Add a suggestion to start from a synced master branch?

Done.

**Comment by nikhita** [comment]
>  The one thing to possibly add is how to pick the commit hash (single commit, vs reverting an entire merge commit).

Done.

**Comment by cblecker** [comment]
/approve
/hold cancel

Thanks @nikhita 🚀

**Pull request merged**